sdk/include/platform/sunburst/platform-gpio.hh

```diff
@@ -136,8 +294,15 @@ struct SonataGPIO
 	/**
 	 * Returns the state of the joystick.
 	 */
-	SonataJoystick read_joystick() volatile
+	JoystickValue read_joystick() volatile
 	{
-		return static_cast<SonataJoystick>(input & 0x1f);
+		return {static_cast<JoystickDirection>(input & Inputs::Joystick)};
 	}
 };
+
+using SonataGpioRaspberryPiHat = SonataGpioBase<0x0FFF'FFFF>;
```

Comment:
Perhaps these should be structs inheriting from their current definitions and providing static factory methods for the singletons on Sonata.  I'm content to leave that for a later PR, tho'.